internal/credential/vault/repository_credential_library.go

```diff
@@ -207,24 +320,221 @@ func (r *Repository) UpdateCredentialLibrary(ctx context.Context, scopeId string
 	return returnedCredentialLibrary, rowsUpdated, nil
 }
 
+type changeFunc func(ctx context.Context, w db.Writer) (*oplog.Message, error)
+
+func calcMappingOverrideChange(op errors.Op, ol, nl *CredentialLibrary, dbMask []string) (changeFunc, error) {
```

Comment:
This function is very dense and difficult (at least for me) to grok. I think a few things could be done to help:

- add some function docs and some inline comments
- potentially break out the `insertFn`, `deleteFn` `updateFn` functions so they can be documented independently
- use some more descriptive variable names, there are a lot of short but similar variable names in here, `om`, `mo`, `nm`, `nmp`, `dmp`